server/src/index.ts

```diff
@@ -1,12 +1,73 @@
+import path from 'path';
+import fs from 'fs';
+
 import Koa from 'koa';
 import Router from 'koa-router';
 import dotenv from 'dotenv';
+import serve from 'koa-static';
 
 dotenv.config();
 const app = new Koa();
 const router = new Router();
 
-// default error handler (수정 예정)
+const STATIC_DIR = path.join(__dirname, 'public');
+
+app.use(router.routes());
+app.use(serve(STATIC_DIR));
+
+/**
+ * @desc
+ * hmr 과 유사한 효과를 낼 수 있게 해주는 미들웨어 (코드 분할 예정)
+ * 클라이언트단 번들 결과물 변경시 SSE 를 통해 브라우저에 신호를 전송하여 리로드하게 합니다. (신호 감지 후 리로드하는 로직은 클라이언트 쪽에 있습니다)
+
+  원리를 정리하면 아래와 같음
+  1. 프론트엔드 작업 공간에서 무언가 코드를 작성
+  2. 프론트엔드측 webpack watch 모드가 변경을 감지하여 리빌드
+  3. koa 서버에서 빌드 디렉터리의 변경을 감지하여 브라우저에 신호 보냄 (SSE)
+  4. 해당 신호를 client/app/entries/common.js 에서 해당 신호를 받아 브라우저 리로딩  (리로딩하는 이유 - 서버에 데이터 재요청을 보내게 되어 변경된 데이터가 적용될 수 있게 함)
+ */
+if (process.env.NODE_ENV !== 'production') {
+  app.use(async (ctx, next) => {
+    if (ctx.path !== '/sse') {
+      await next();
+      return;
+    }
+
+    ctx.request.socket.setTimeout(0);
+    ctx.req.socket.setNoDelay(true);
+    ctx.req.socket.setKeepAlive(true);
+
+    ctx.set({
+      'Content-Type': 'text/event-stream',
+      'Cache-Control': 'no-cache',
+      Connection: 'keep-alive',
+    });
+
+    const { PassThrough } = await import('stream');
```

Comment:
```suggestion
    const { PassThrough } = await import('stream');
```
모든 요청에 대해서 해당 미들웨어를 통과하면서, 매 요청마다 stream 에서 PassThrough 스트림이 import 될텐데 dynamic import 로 작성하신 이유 있으신가요?
최상단에 import 하는게 좋을 듯 합니다.

Reply:
서버측 env와 클라이언트측 env가 모두 production이 아닐 때, /sse 로 요청이 들어온 경우만 연결됩니다.

Reply:
따봉 말고 approve